text/0085-api-keys.md

```diff
@@ -0,0 +1,547 @@
+| indexes.add | Provides access to `POST` `/indexes`. **⚠️ A newly created `index` is added to the `indexes` array for the API key making the operation and all others keys having `[*]` as a value for `indexes`**. |
+| indexes.get | Provides access to `GET` `/indexes` and `/indexes/:authorizedIndexes`. **⚠️Non-authorized `indexes` are omitted from the response on `/indexes`**. |
+| indexes.update | Provides access to `PUT` `/indexes/:authorizedIndexes`. |
+| indexes.delete | Provides access to `DELETE` `/indexes/:authorizedIndexes`. |
+| tasks.get | Provides access to `GET` `/tasks`. **⚠️Non-authorized `indexes` are omitted from the response on `/tasks`**. Also add access to `GET` `/indexes/:authorizedIndexes/tasks` routes. |
+| settings.get | Provides access to `GET` `/indexes/:authorizedIndexes/settings` and `/indexes/:authorizedIndexes/settings/*` routes. |
+| settings.update | Provides access to `POST / DELETE` `/indexes/:authorizedIndexes/settings` and `/indexes/:authorizedIndexes/settings/*` routes. |
+| stats.get | Provides access to `GET` `/stats/`. **⚠️Non-authorized `indexes` are omitted from the response on `/stats`**. Also add access to `GET` `/indexes/:authorizedIndexes/stats`. |
+| dumps.create | Provides access to `POST` `/dumps` route. **As dumps are not scoped by indexes, a restriction on `indexes` does not affect this action.** |
+| dumps.get | Provides access to `GET` `/dumps/:dumpUid` route. **As dumps are not scoped by indexes, a restriction on `indexes` does not affect this action.** |
+| version | Provides access to `GET` `/version` route.
+
+---
+
+#### **As `Anna 👩`, I want to create an `API key` for `Mark 👨🏻` client-code, so that he can index some documents into MeiliSearch**
+
+##### Request Definition
+
+`POST` - `/keys`
+
+##### Headers
+
+```
+"Authorization: Bearer :masterKey"
+"Content-Type: application/json"
+```
+
+##### Body Payload
+
+```json
+{
+    "description": "Indexing Products API key",
+    "actions": [
+        "documents.add"
+    ],
+    "indexes": ["products"],
+    "expiresAt": "2021-11-13T00:00:00Z"
+}
+```
+
+##### Response
+
+`201 Created`
+
+```json
+{
+    "description": "Indexing Products API key",
+    "key": "d0552b41536279a0ad88bd595327b96f01176a60c2243e906c52ac02375f9bc4",
+    "actions": [
+        "documents.add"
+    ],
+    "indexes": ["products"],
+    "expiresAt": "2021-11-13T00:00:00Z",
+    "createdAt": "2021-11-12T10:00:00Z",
+    "updatedAt": "2021-11-12T10:00:00Z"
+}
+```
+
+##### Requirements
+
+- `actions` is mandatory and should be an array of valid `actions`.
+- `indexes` is mandatory and should be an array of string.
+- `expiresAt` is mandatory and must be a valid `ISO 8601` datetime in the future or `null`.
+- If set, `description` should be a string or `null`.
```

Comment:
Is it possible to have an additional field that contains an unique name for the key? In order to be able to fetch the apiKey without having to watch with human eyes at which index the key is: 

### with name

```js 
const keys = client.getKeys()
const privateKey = keys.find(name => name == "default_private")
```

### Without name

```js
const keys = client.getKeys()
console.log(keys) // look at the description and either count at which index the right key is, or copy paste it from the terminal
```

Reply:
Yes @bidoubiwa! We could also have a filter system on the indexes and action fields, also a parameter to search in the description field.

This may be in a future iteration.

Reply:
Thanks :)